cmd/duffle/bundle_remove.go

```diff
@@ -18,80 +18,93 @@ const bundleRemoveDesc = `Remove a bundle from the local storage.
 		RunE: func(cmd *cobra.Command, args []string) error {
-			bname := args[0]
+			remove.bundleRef = args[0]
+			remove.home = home.Home(homePath())
+			return remove.run()
+		},
+	}
+	cmd.Flags().StringVar(&remove.versions, "version", "", "A version or SemVer2 version range")
 
-			h := home.Home(homePath())
-			index, err := repo.LoadIndex(h.Repositories())
-			if err != nil {
-				return err
-			}
+	return cmd
+}
 
-			vers, ok := index.GetVersions(bname)
-			if !ok {
-				fmt.Fprintf(w, "Bundle %q not found. Nothing deleted.", bname)
-				return nil
-			}
+func (rm *bundleRemoveCmd) run() error {
+	index, err := repo.LoadIndex(rm.home.Repositories())
+	if err != nil {
+		return err
+	}
 
-			// If versions is set, we short circuit and only delete specific versions.
-			if versions != "" {
-				fmt.Fprintln(w, "Only deleting versions")
-				matcher, err := semver.NewConstraint(versions)
-				if err != nil {
-					return err
-				}
-				deletions := []repo.BundleVersion{}
-				for _, ver := range vers {
-					if ok, _ := matcher.Validate(ver.Version); ok {
-						fmt.Fprintf(w, "Version %s matches constraint %q\n", ver, versions)
-						deletions = append(deletions, ver)
-						index.DeleteVersion(bname, ver.Version.String())
-						// If there are no more versions, remove the entire entry.
-						if vers, ok := index.GetVersions(bname); ok && len(vers) == 0 {
-							index.Delete(bname)
-						}
-
-					}
-				}
-				// We can skip writing the file if there is nothing to delete.
-				if len(deletions) == 0 {
-					return nil
-				}
-				if err := index.WriteFile(h.Repositories(), 0644); err != nil {
-					return err
+	vers, ok := index.GetVersions(rm.bundleRef)
+	if !ok {
+		fmt.Fprintf(rm.out, "Bundle %q not found. Nothing deleted.", rm.bundleRef)
+		return nil
+	}
+
+	// If versions is set, we short circuit and only delete specific versions.
+	if rm.versions != "" {
+		fmt.Fprintln(rm.out, "Only deleting versions")
+		matcher, err := semver.NewConstraint(rm.versions)
+		if err != nil {
+			return err
+		}
+		deletions := []repo.BundleVersion{}
+		for _, ver := range vers {
+			if ok, _ := matcher.Validate(ver.Version); ok {
+				fmt.Fprintf(rm.out, "Version %s matches constraint %q\n", ver, rm.versions)
+				deletions = append(deletions, ver)
+				index.DeleteVersion(rm.bundleRef, ver.Version.String())
+				// If there are no more versions, remove the entire entry.
+				if vers, ok := index.GetVersions(rm.bundleRef); ok && len(vers) == 0 {
+					index.Delete(rm.bundleRef)
 				}
-				deleteBundleVersions(deletions, index, h, w)
-				return nil
-			}
 
-			// If no version was specified, delete entire record
-			if !index.Delete(bname) {
-				fmt.Fprintf(w, "Bundle %q not found. Nothing deleted.", bname)
-				return nil
 			}
-			if err := index.WriteFile(h.Repositories(), 0644); err != nil {
-				return err
-			}
-
-			deleteBundleVersions(vers, index, h, w)
+		}
+		// We can skip writing the file if there is nothing to delete.
```

Comment:
_skip writing [the index] file_?
It took me a while to understand what file it references.

Reply:
yea agreed. I deleted the comment. I think it's obvious from the code that what is happening is if you don't have anything to delete, then don't touch anything and return